lib/tasks/integration.rake

```diff
@@ -1,144 +1,218 @@
+  task :integration => :prepare do
+    Rake::Task['integrate:run_tests'].invoke(test_commands[:integration])
+  end
 
-  if junit.present?
-    puts 'WARNING: tmp/junit is not empty'
-    puts junit
-    abort 'Will not continue, tmp/junit is not empty'
+  task :swagger => :prepare do
+    Rake::Task['integrate:run_tests'].invoke(test_commands[:swagger])
   end
 
-  total_time = 0
+  task :frontend => :prepare do
+    Rake::Task['integrate:run_tests'].invoke(test_commands[:frontend])
+  end
 
-  results = tasks.map do |task|
+  task :functional => :prepare do
+    Rake::Task['integrate:run_tests'].invoke(test_commands[:functional])
+  end
 
-    command = nil
-    result = report.execute(task, env: {RAILS_ENV: Rails.env}) do |cmd|
-      banner.("BEGIN: #{cmd}")
-      command = cmd
-    end
+  task :main_suite => :prepare do
+    Rake::Task['integrate:run_tests'].invoke(test_commands[:main_suite])
+  end
 
-    summary << banner.("FINISH (#{result.success? ? success : failure}): #{command} in #{'%.1fs' % result.time}")
-    total_time += result.time
+  task :cucumber_non_tagged => :prepare do
+    Rake::Task['integrate:run_tests'].invoke(test_commands[:cucumber_non_tagged])
+  end
+
+  task :cucumber_for_categories => :prepare do
+    Rake::Task['integrate:run_tests'].invoke(test_commands[:cucumber_for_categories])
+  end
 
-    result.success?
+  task :license_checks => :prepare do
+    Rake::Task['integrate:run_tests'].invoke(test_commands[:license_checks])
   end
 
-  banner.("SUMMARY: in #{'%.1fs' % total_time}\n\t#{summary.join("\n\t")}")
+  task :percy => :prepare do
+    Rake::Task['integrate:run_tests'].invoke(test_commands[:percy])
+  end
+
+
+  desc 'Runs subset of full test suite, in parallel (1/8)'
+  task :parallel_1 => %i[verify_empty_reports_folder prepare cucumber_javascript rspec]
+
+  desc 'Runs subset of full test suite, in parallel (2/8)'
+  task :parallel_2 => %i[verify_empty_reports_folder prepare integration]
+
+  desc 'Runs subset of full test suite, in parallel (3/8)'
+  task :parallel_3 => %i[verify_empty_reports_folder prepare integration]
 
-  abort "some tasks failed, exitting" unless results.all?
+  desc 'Runs subset of full test suite, in parallel (4/8)'
+  task :parallel_4 => %i[verify_empty_reports_folder prepare functional main_suite]
 
-  if ENV['COVERAGE']
-    FileUtils.cp(Dir["#{Dir.tmpdir}/codeclimate-test-coverage-*"],
-                 Rails.root.join('tmp', 'codeclimate').tap(&:mkpath))
+  desc 'Runs subset of full test suite, in parallel (5/8)'
+  task :parallel_5 => %i[verify_empty_reports_folder prepare cucumber_non_tagged]
+
+  desc 'Runs subset of full test suite, in parallel (6/8)'
+  task :parallel_6 => %i[verify_empty_reports_folder prepare cucumber_for_categories]
+
+
+
+  desc 'Runs the whole continuous integration test suite'
+  task :parallel, [:log] do |t, args|
+
+    banner = print_banner_around
+
+    time = Benchmark.measure do
+
+      Rake::Task['integrate:parallel_1'].invoke
```

Comment:
I think it should be possible to get all the defined rake tasks and filter, instead of maintaining it by hand.

```ruby
Rake::Task.tasks.select { |task| task.name =~ /^integrate:parallel_/ }
````